src/components/password.components.js

```diff
@@ -0,0 +1,97 @@
+import $ from 'jquery';
+
+export default class PasswordInput {
+    constructor(queryString, valid = false, disabled = false, required = true) {
+        this.div = $(queryString);
+        this.input = $(`${queryString} input[type="password"]`);
+        this.icon = $(`${queryString} input[type="password"]+i`);
+
+        this.valid = valid;
+        this.disabled = disabled;
+        this.required = required;
+
+        this.checkQuery(this.div, this.input, this.icon);
+        this.addClickEvent();
+    }
+
+    checkQuery(div, input, icon) {
+        if (div.length === 0) {
+            throw new Error('Failed to query div');
+        }
+
+        if (input.length === 0) {
+            throw new Error('Failed to query input password');
+        }
+
+        if (icon.length === 0) {
+            throw new Error('Failed to query icon');
+        }
+    }
+
+    addClickEvent() {
+        this.icon.on('click', () => {
+            if (this.input.attr('type') === 'password') {
+                this.input.attr('type', 'text').trigger('focus');
+                this.icon.removeClass('fa-eye-slash').addClass('fa-eye');
+            } else {
+                this.input.attr('type', 'password').trigger('focus');
+                this.icon.removeClass('fa-eye').addClass('fa-eye-slash');
+            }
+        });
+    }
+
+    getValue() {
+        return this.input.val();
+    }
+
+    setValue(string) {
```

Comment:
This comportament is wrong
![image](https://user-images.githubusercontent.com/23269934/163169947-68aa9a5f-75e3-420e-b462-e7a31ccec601.png)